### ISSUE TITLE
RFE: Add a container registry

### ISSUE THREAD

**Issue by christianh814** [RedHatOfficial/ocp4-helpernode#9]
Add the ability to host a container registry. This can be helpful for disconnected installs or any ancillary container images one may want to host.

Here's some steps for a [registry setup](https://github.com/christianh814/blogs/tree/master/docs/openshift-4.2-restricted-network-install#registry-set-up)

After those steps you can create a systemd service unit file to start it on boot

```
#
# Copy (chmod 664) to /etc/systemd/system/poc-registry.service
#
# systemctl daemon-reload
# systemctl start poc-registry
# systemctl status poc-registry
# systemctl enable poc-registry
#

[Unit]
Description=OpenShift POC HTTP for PXE Config
After=network.target syslog.target

[Service]
Type=simple
TimeoutStartSec=5m
ExecStartPre=-/usr/bin/podman rm "poc-registry"

ExecStart=/usr/bin/podman run   --name poc-registry -p 5000:5000 \
                                -v /opt/registry/data:/var/lib/registry:z \
                                -v /opt/registry/auth:/auth:z \
                                -e "REGISTRY_AUTH=htpasswd" \
                                -e "REGISTRY_AUTH_HTPASSWD_REALM=Realm" \
                                -e REGISTRY_AUTH_HTPASSWD_PATH=/auth/htpasswd \
                                -v /opt/registry/certs:/certs:z \
                                -e REGISTRY_HTTP_TLS_CERTIFICATE=/certs/domain.crt \
                                -e REGISTRY_HTTP_TLS_KEY=/certs/domain.key \
                                docker.io/library/registry:2

ExecReload=-/usr/bin/podman stop "poc-registry"
ExecReload=-/usr/bin/podman rm "poc-registry"
ExecStop=-/usr/bin/podman stop "poc-registry"
Restart=always
RestartSec=30

[Install]
WantedBy=multi-user.target
```

**Comment by flickerfly** [comment]
Yes, this would be very useful.

**Comment by christianh814** [comment]
Just got this note about the NFS provisioner from someone working on a disconnected install

> So.  Helpernode + vsphere-automation + disconnected means the image for the nfs-provisioner is not available.  While trying to add it to the local registry I ran into a manifest version issue...   the image is v1 but the registry is v2.   Anyhow, looks like skopeo can convert the manifest for you using --format=v2s2
https://github.com/containers/skopeo/issues/959

**Comment by Prajyot-Parab** [comment]
@christianh814 Please take a look at #86

**Comment by christianh814** [comment]
@Prajyot-Parab 

Thanks for this! I'll take a look later today :+1:

**Comment by christianh814** [comment]
This is now in the `devel` branch. Will merge to the `master` branch next release.

**Comment by christianh814** [comment]
Closed by #93
